### DISCUSSION TITLE
Node Version

### DISCUSSION THREAD

**Discussion by Bajweb** [holger1411/understrap#1566] [General]
Hello, please can you tell me what is the lowest version of node that I can use with understrap?
Thank you for your replay.

Alex

**Comment by woodwardtw** [comment]
The package json file says   

```
"engines": {
    "node": ">=14"
  },
```

so I'm reading that as recommending greater than 14.

**Reply by Bajweb** [reply]
Thank you

**Reply by bacoords** [reply]
That is correct, though I would recommend 16 if you can swing it, I feel like we've heard issues on 14.

**Reply by Bajweb** [reply]
> That is correct, though I would recommend 16 if you can swing it, I feel like we've heard issues on 14.

Thank you so much,
I installed the latest node version. Changes that I provide on /src/sass/theme/_theme.scss file doesn't appears on website. Terminal is running ok, no errors, even the code is compile and minify - I can see it in /css/theme.min.css file. Also the changes in php files like header.php or footer.php are applying too, but style( scss ) and JS doesn't. What the problem may be?
I don't use child theme.

Terminaal

understrap@1.1.0 css
npm-run-all css-compile css-postcss css-minify

understrap@1.1.0 css-compile
sass --style expanded --source-map --embed-sources --no-error-css --quiet src/sass/theme.scss:css/theme.css src/sass/custom-editor-style.scss:css/custom-editor-style.css

understrap@1.1.0 css-postcss
postcss --config src/build/postcss.config.js --replace "css/.css" "!css/.rtl*.css" "!css/*.min.css"

understrap@1.1.0 css-minify
cleancss -O1 --format breakWith=lf --with-rebase --source-map --source-map-inline-sources --output css/ --batch --batch-suffix ".min" "css/.css" "!css/.min.css" "!css/rtl.css"

[nodemon] clean exit - waiting for changes before restart
[Browsersync] File event [change] : css\custom-editor-style-bootstrap4.min.css
[Browsersync] File event [change] : css\custom-editor-style.min.css
[Browsersync] File event [change] : css\theme-bootstrap4.min.css
[Browsersync] File event [change] : css\theme.min.css

Thanks Alex

**Reply by woodwardtw** [reply]
I'm no expert but I had a similar problem. In package.json I ended up changing 
`"watch-run-css": "nodemon --watch src/sass/ --ext scss --exec \"npm-run-all css\"",`
to
` "watch-run-css": "nodemon --watch src/sass/ --ext scss --exec \"npm-run-all css css-bs4\"",`

**Reply by Bajweb** [reply]
> I'm no expert but I had a similar problem. In package.json I ended up changing `"watch-run-css": "nodemon --watch src/sass/ --ext scss --exec \"npm-run-all css\"",` to ` "watch-run-css": "nodemon --watch src/sass/ --ext scss --exec \"npm-run-all css css-bs4\"",`

Woooow It works for me now!!! That's great , thank you so much Tom...
YOU ARE EXPERT !!!

JS still doesn't work, do you know how to fix that?

Thank's Alex

**Reply by woodwardtw** [reply]
You can run `dist` manually to get everything to run. But I think maybe adding `js-bs4` to `watch-run-js` would do it. I haven't tested that though.

**Reply by Bajweb** [reply]
Unfortunately js-bs4 doesn't work... anyway thank you for your time and effort.

**Reply by woodwardtw** [reply]
I didn't end up needing to change anything. I think you want to add your custom js to src>js>custom-javascript.js 

Take note that I believe it removes console.log messages when the build runs so you may not see it as functional if that's what you're looking for.

**Reply by bacoords** [reply]
By default `npm run watch` just compiles for Bootstrap 5 assets (instead of compiling everything for BS4 and BS5 every time). If you're planning on using Bootstrap4 for your next Understrap project, you can use the other commands: `npm run dist` to compile everything OR you can modify the `npm run watch` commands to compile for BS4 (per @woodwardtw ).

Full documentation: https://github.com/understrap/understrap#bootstrap-4-legacy-build-process

**Reply by Bajweb** [reply]
Thank you guys, css works but JS still doesn't. I added js-bs4 in "watch-run-js". The npm compile the code from src>js>custom-javascript.js I see it in js>theme.min.js but not in frontend ( in console )

**Reply by Bajweb** [reply]
I try now to add "js-bs4 css-bs4" to "watch-bs" and everything works JS an CSS, thanks a lot...
Alex
